athenaStore.js

```diff
@@ -0,0 +1,67 @@
+const AWS = require('aws-sdk');
+const uuid = require('uuid');
+const table = require('table');
+
+const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));
+
+class AthenaLetterStore {
+  constructor({dbName, s3Path, pollInterval}) {
+    this.dbName = dbName;
+    this.s3Path = s3Path;
+    this.pollInterval = pollInterval;
+
+    this.athena = new AWS.Athena();
+  }
+
+  async runQuery(query) {
+    console.log("Running Athena query:", query);
+    const queryID = uuid.v4();
+
+    const startRes = await this.athena.startQueryExecution({
+      QueryString: query,
+      ResultConfiguration: {
+        OutputLocation: this.s3Path,
+      },
+      ClientRequestToken: queryID,
+      QueryExecutionContext: {
+        Database: this.dbName,
+      }
+    }).promise();
+
+    let res;
+    while (true) {
+      res = await this.athena.getQueryExecution({
+        QueryExecutionId: startRes.QueryExecutionId,
+      }).promise();
+
+      console.log("State is: ", res.QueryExecution.Status.State);
+      if (['SUCCEEDED', 'FAILED', 'CANCELLED'].includes(res.QueryExecution.Status.State)) {
+        break;
+      }
+
+      await sleep(this.pollInterval);
```

Comment:
Perhaps this while loop should time out eventually. As is, this could run forever if an Athena query gets stuck (not sure how likely that is though).

Reply:
Yep, will be caught by the timeout.